File: dotnet/Lexy/Lexy.RunTime/RunTime/ExecutionContext.cs
using System;
using Microsoft.Extensions.Logging;

namespace Lexy.RunTime.RunTime
{
    public class ExecutionContext : IExecutionContext
    {
        private readonly ILogger<ExecutionContext> logger;

        public ExecutionContext(ILogger<ExecutionContext> logger)
        {
            this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
        }

        public void LogDebug(string message)
        {
            logger.LogDebug(message);
        }

        public void LogVariable<T>(string name, T value)
        {
            logger.LogDebug("  {Name}: {Value}", name, value);
        }
    }
}

File: typescipt/src/runTime/executionContext.ts


export class ExecutionContext extends IExecutionContext {
   private readonly ILogger<ExecutionContext> logger;

   constructor(logger: ILogger<ExecutionContext>) {
     this.logger = logger ?? throw new Error(nameof(logger));
   }

   public logDebug(message: string): void {
     logger.LogDebug(message);
   }

   public logVariable<T>(name: string, value: T): void {
     logger.LogDebug(` {Name}: {Value}`, name, value);
   }
}
